<a href="https://colab.research.google.com/github/FGalvao77/Machine-Learning/blob/main/Oficina_PLN.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Bibliotecas necessárias

In [1]:
! pip install nltk

In [2]:
import nltk
from sklearn.feature_extraction.text import TfidfVectorizer
import pandas as pd

### Pacotes necessários

In [3]:
nltk.download('stopwords')
nltk.download('punkt')

[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Unzipping corpora/stopwords.zip.
[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Unzipping tokenizers/punkt.zip.


True

Use a variável texto a seguir para identificar as stop words
e após isso aplique o cáculo do **TF-IDF** neste mesmo texto e observe os scores.

**OBS:** Para realizar essa tarefa utilize os módulos estudados na aula

In [4]:
texto = 'Um carro azul seguia rapidamente em uma rodovia, e ao passar por um buraco, o carro furou o pneu, e o motorista desceu do carro azul'

In [5]:
# instanciando o módulo
tf_idf = TfidfVectorizer()

In [6]:
# utilizando o fit transform para termos uma matriz
vetor = tf_idf.fit_transform([texto])

# imprimindo o vetor
print(vetor)

  (0, 5)	0.1796053020267749
  (0, 4)	0.1796053020267749
  (0, 8)	0.1796053020267749
  (0, 10)	0.1796053020267749
  (0, 7)	0.1796053020267749
  (0, 2)	0.1796053020267749
  (0, 11)	0.1796053020267749
  (0, 9)	0.1796053020267749
  (0, 0)	0.1796053020267749
  (0, 13)	0.1796053020267749
  (0, 16)	0.1796053020267749
  (0, 6)	0.1796053020267749
  (0, 12)	0.1796053020267749
  (0, 14)	0.1796053020267749
  (0, 1)	0.3592106040535498
  (0, 3)	0.5388159060803247
  (0, 15)	0.3592106040535498


In [7]:
# transformando a matriz em um array
vetor = vetor.todense()
print(vetor)

[[0.1796053  0.3592106  0.1796053  0.53881591 0.1796053  0.1796053
  0.1796053  0.1796053  0.1796053  0.1796053  0.1796053  0.1796053
  0.1796053  0.1796053  0.1796053  0.3592106  0.1796053 ]]


In [8]:
# utilizando a função ".get_feature_names()"
# para mapear as palavras
nomes = tf_idf.get_feature_names()

print(nomes)

['ao', 'azul', 'buraco', 'carro', 'desceu', 'do', 'em', 'furou', 'motorista', 'passar', 'pneu', 'por', 'rapidamente', 'rodovia', 'seguia', 'um', 'uma']


In [9]:
# transformando o resultado em dataframe
df = pd.DataFrame(vetor, columns=nomes)

# visualizando o dataframe criado
print(df)

         ao      azul    buraco  ...    seguia        um       uma
0  0.179605  0.359211  0.179605  ...  0.179605  0.359211  0.179605

[1 rows x 17 columns]


> Perceba que, as palavras `azul` e `um` foram as que tiveram igualmente maior peso (0.359211) e as demais tamabém tiveram igualmente mesmo peso (0.179605)